libraries/ms-lib/mslib.mk

```diff
@@ -1,19 +1,19 @@
-mslib = libraries/ms-lib
+MSLIB := libraries/ms-lib
 
-mssrcdir = $(mslib)/src/$(devicefamily)
-msincdir = $(mslib)/inc
-msobjdir = $(mslib)/obj
+MSSRC_DIR := $(MSLIB)/src/$(DEVICE_FAMILY)
+MSINC_DIR := $(MSLIB)/inc
+MSOBJ_DIR := $(MSLIB)/obj
 
-mssrc = $(notdir $(wildcard $(mssrcdir)/*.c)) 
-msobj = $(addprefix $(msobjdir)/,$(mssrc:.c=.o))
+MSSRC := $(notdir $(wildcard $(MSSRC_DIR)/*.c)) 
+MSOBJ := $(addprefix $(MSOBJ_DIR)/,$(MSSRC:.c=.o))
 
-msCFLAGS = -g -O2 -Wall -Werror -pedantic -Wno-unused-variable \
-					 $(ARCH) $(LIB) -I$(mslib) \
+MSCFLAGS := -g -O2 -Wall -Werror -pedantic -Wno-unused-variable \
+					 $(ARCH) $(LIB) -I$(MSLIB) \
 					 -ffreestanding -nostdlib
 
-$(mslib)/mslib.a: $(msobj)
-	@$(AR) -r $@ $(msobj)
+$(MSLIB)/mslib.a: $(MSOBJ)
```

Comment:
This is the one thing I don't think is possible. You need a build.mk/rules.mk file per library (we can have a template for this) but because file structure is different and the object directories and cflags are likely different for each library so you can't have a static rule because so far as I can tell make doesn't let you do things like indexing into a list to get corresponding entries. If you know of a way to do this feel free to correct me or let me know but based on how I understand make works this is not possible.